src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -0,0 +1,397 @@
+   */
+  public OsisoftJsonPayload(int comType) {
+
+    int byteSizePerDataPoint = 80;
+    int byteSizeForStartAndEnd = 200;
+    int maxByteSize = (MAX_DATA_POINTS * byteSizePerDataPoint) + byteSizeForStartAndEnd;
+    payload = new PreAllocatedStringBuilder(maxByteSize);
+    status = PAYLOAD_NOT_STARTED;
+    communicationType = comType;
+
+    startPayload();
+
+    // set up array for tags encountered
+    int lowestSeen = TagInfoManager.getLowestTagIdSeen();
+    int highestSeen = TagInfoManager.getHighestTagIdSeen();
+
+    if (highestSeen != TagConstants.UNINIT_INT_VAL) {
+      tagIndexOffset = lowestSeen;
+      tagsEncounteredArraySize = highestSeen - lowestSeen + 1;
+      tagNamesEncountered = new boolean[tagsEncounteredArraySize];
+      tagPayloadArr = new TagPayload[tagsEncounteredArraySize];
+    } else {
+      Logger.LOG_SERIOUS(
+          "Unable to initialize tags list."
+              + " It is possible no tags have been added to this device.");
+      tagNamesEncountered = new boolean[0];
+    }
+  }
+
+  /**
+   * Set the value of the tagId's index to true if it is being 'added' to this array
+   *
+   * @param tagId Id of the tag
+   */
+  private void addToTagsEncounteredArray(int tagId) {
+    int index = tagId - tagIndexOffset;
+    tagNamesEncountered[index] = true;
+  }
+
+  /**
+   * Return if the specified tag ID has been encountered
+   *
+   * @param tagId tag ID
+   * @return true if tag ID has been encountered
+   */
+  private boolean checkIfTagIdEncountered(int tagId) {
+    int index = tagId - tagIndexOffset;
+    return tagNamesEncountered[index];
+  }
+
+  /**
+   * Get the payload as a string
+   *
+   * @return payload string
+   */
+  public String getPayload() {
+    return payload.toString();
+  }
+
+  /**
+   * This function starts a new payload with the objects payload settings.
+   *
+   * <p>The status of the payload will be updated to show that the header information has been
+   * added.
+   */
+  public void startPayload() {
+
+    payload.clearString();
+    tagNamesEncountered = new boolean[tagsEncounteredArraySize];
+    tagPayloadArr = new TagPayload[tagsEncounteredArraySize];
+    payloadStartTimestamp = 0;
+    dataPointsEncounteredCount = 0;
+    String startOfPayload;
+
+    switch (communicationType) {
+      case OsisoftConfig.omf:
+        startOfPayload = PayloadBuilder.startOMFDataMessage();
+        break;
+      case OsisoftConfig.piwebapi:
+        startOfPayload = PayloadBuilder.startBatchOldFormat();
+        break;
+      default:
+        Logger.LOG_SERIOUS(comsErrMsg);
+        startOfPayload = "no communication type";
+        break;
+    }
+
+    boolean didAppendToPayload = payload.append(startOfPayload);
+    if (didAppendToPayload) {
+      status = PAYLOAD_HEADER_COMPLETE;
+    } else {
+      Logger.LOG_CRITICAL("An error occurred while appending the start of a payload.");
+    }
+  }
+
+  /**
+   * Append a data point to the payload
+   */
```

Comment:
missing @ param

Reply:
can you start adding in the function name as well, these are a pain to find once line numbers change